daemon/checkpoint.go

```diff
@@ -0,0 +1,50 @@
+package daemon
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/docker/docker/runconfig"
+)
+
+// ContainerCheckpoint checkpoints the process running in a container with CRIU
+func (daemon *Daemon) ContainerCheckpoint(name string, opts *runconfig.CriuConfig) error {
+	container, err := daemon.Get(name)
+	if err != nil {
+		return err
+	}
+	if !container.IsRunning() {
+		return fmt.Errorf("Container %s not running", name)
+	}
+
+	if opts.ImagesDirectory == "" {
+		opts.ImagesDirectory = filepath.Join(container.Root, "criu.image")
+		if err := os.MkdirAll(opts.ImagesDirectory, 0755); err != nil && !os.IsExist(err) {
+			return err
+		}
+	}
+
+	if opts.WorkDirectory == "" {
+		opts.WorkDirectory = filepath.Join(container.Root, "criu.work")
+		if err := os.MkdirAll(opts.WorkDirectory, 0755); err != nil && !os.IsExist(err) {
+			return err
+		}
+	}
+
+	if err := daemon.Checkpoint(container, opts); err != nil {
+		return fmt.Errorf("Cannot checkpoint container %s: %s", name, err)
+	}
+
+	container.SetCheckpointed(opts.LeaveRunning)
```

Comment:
Should we produce an event in here as these don't get stop event but container is not active any more. And we do have an event for restore.